opm/simulators/wells/WellState.cpp

```diff
@@ -424,7 +424,9 @@ void WellState<Scalar>::init(const std::vector<Scalar>& cellPressures,
         }
     }
 
-    updateWellsDefaultALQ(wells_ecl, summary_state);
+    // ALQ state needs to be defined for all wells, otherwise alqstate.packSize() will be different
+    // on each process, and the comm.sum call in communicateGroupRates will fail.
+    updateWellsDefaultALQ(schedule.getWells(report_step), summary_state);
```

Comment:
I appreciate the sentiment and we clearly need to fix this.  However [`Schedule::getWells()`](https://github.com/OPM/opm-common/blob/e85c8b8c8abcae34279de425093754a497b6a24f/opm/input/eclipse/Schedule/Schedule.cpp#L1047-L1057) is an expensive function, so if we could make this less expensive then that would be preferable.  Perhaps a new (private) overload of `WellState::updateWellsDefaultALQ()` of the form
```C++
template <typename Scalar>
void WellState<Scalar>::updateWellsDefaultALQ(const Schedule& schedule,
                                              const int report_step,
                                              const SummaryState& summary_state)
{
    const auto wells = schedule.wellNames(report_step);
    for (const auto& wname : wells) {
        const auto& well = schedule.getWell(wname, report_step);
        if (! well.isProducer()) {
            continue;
        }
        const auto alq = well.alq_value(summary_state);
        this->alq_state.update_default(wname, alq);
    }
}
```
since [`Schedule::getWell()`](https://github.com/OPM/opm-common/blob/e85c8b8c8abcae34279de425093754a497b6a24f/opm/input/eclipse/Schedule/Schedule.cpp#L1082-L1084) does not create copies of the `Well` objects?

Reply:
Good point, I didn't think of the cost. Maybe just change the signature to what you propose above and update the single other call to the method (in BlackoilWellModel_impl.hpp)? Might bring us one step closer to handling switching in ACTIONX also (maybe - didn't dig into it..)